scope.go

```diff
@@ -269,108 +270,155 @@ func (s *scope) reportRegistryWithLock() {
 		}
 		s.cachedReporter.Flush()
 	}
-	s.registry.RUnlock()
 }
 
 func (s *scope) Counter(name string) Counter {
+	if c, ok := s.counter(name); ok {
```

Comment:
i think the sanitizer.Name needs to be called before we do this check here? maybe the argument name in `counter` should be called `sanitizedName` to make it clearer as well?

Reply:
yeah, nice catch. thanks!